crates/host/src/wasmbus/mod.rs

```diff
@@ -1101,12 +1179,70 @@ impl Lattice {
                 instances: HashMap::default(),
             });
         if let hash_map::Entry::Vacant(entry) = instances.entry(link_name.clone()) {
-            debug!(?path, "spawn provider process");
-            let child = process::Command::new(&path)
+            let id = Ulid::new();
+            let async_nats::ServerInfo { host, port, .. } = self.nats.server_info();
+            let invocation_seed = self
+                .cluster_key
+                .seed()
+                .context("cluster key seed missing")?;
+            let links = self.links.read().await;
+            let link_definitions: Vec<_> = links
+                .values()
+                .filter(|ld| ld.provider_id == claims.subject && ld.link_name == link_name)
+                .collect();
+            let data = serde_json::to_vec(&json!({
+                "host_id": self.host_key.public_key(),
+                "lattice_rpc_prefix": "default", // TODO: Support lattice prefix config
+                "link_name": link_name,
+                "lattice_rpc_user_jwt": "", // TODO: Support config
+                "lattice_rpc_user_seed": "", // TODO: Support config
+                "lattice_rpc_url": format!("{host}:{port}"),
+                "lattice_rpc_tls": 0, // TODO: Support config
+                "env_values": {},
+                "instance_id": Uuid::from_u128(id.into()),
+                "provider_key": claims.subject,
+                "link_definitions": link_definitions,
+                "config_json": configuration,
+                "default_rpc_timeout_ms": 2000, // TODO: Support config
+                "cluster_issuers": vec![self.cluster_key.public_key()], // TODO: Support config
+                "invocation_seed": invocation_seed,
+                // TODO: Set `js_domain`
+                // TODO: Set `structured_logging`
+                // TODO: Set `log_level`
+            }))
+            .context("failed to serialize provider data")?;
+
+            debug!(
+                ?path,
+                data = &*String::from_utf8_lossy(&data),
+                "spawn provider process"
+            );
+            let mut child = process::Command::new(&path)
+                .stdin(Stdio::piped())
                 .kill_on_drop(true)
                 .spawn()
                 .context("failed to spawn provider process")?;
-            let id = Ulid::new();
+            let mut stdin = child.stdin.take().context("failed to take stdin")?;
+            stdin
+                .write_all(STANDARD_NO_PAD.encode(&data).as_bytes())
+                .await
+                .context("failed to write provider data")?;
+            stdin
+                .write_all(b"\r\n")
+                .await
+                .context("failed to write newline")?;
+            stdin.shutdown().await.context("failed to close stdin")?;
+
+            let child = spawn(async move {
```

Comment:
Do we want to shadow `child`? We're returning it as part of the `ProviderInstance` below, so we can issue shutdown messages to the provider. If we pass this child handle instead, won't we kill this process, rather than the provider process?